crates/ruff_python_formatter/src/expression/expr_lambda.rs

```diff
@@ -62,7 +83,65 @@ impl FormatNodeRule<ExprLambda> for FormatExprLambda {
             }
         }
 
-        write!(f, [body.format()])
+        // Avoid parenthesizing lists, dictionaries, etc.
+        if is_parenthesize_lambda_bodies_enabled(f.context())
+            && has_own_parentheses(body, f.context()).is_none()
+            && !is_expression_parenthesized(body.into(), comments.ranges(), f.context().source())
+        {
+            match self.layout {
+                ExprLambdaLayout::Default => maybe_parenthesize_expression(
+                    body,
+                    item,
+                    crate::expression::parentheses::Parenthesize::IfBreaksParenthesizedNested,
+                )
+                .fmt(f),
+                ExprLambdaLayout::Assignment => {
+                    fits_expanded(&optional_parentheses(&body.format())).fmt(f)
```

Comment:
Can you try `parenthesized` instead of `optional_parentheses`. 

```suggestion
                    fits_expanded(&parenthesized("(", &body.format(), ")")).fmt(f)
```

Reply:
It looks like this always parenthesizes the body, one example:

```diff
#### Preview changes            
```diff                         
--- Stable                      
+++ Preview                     
@@ -23,8 +23,8 @@               
     pass                       
                                
                                
-x1 = lambda y: 1               
-x2 = lambda y,: 1              
+x1 = lambda y: (1)             
+x2 = lambda y,: (1)            
                                
 # Ignore trailing comma.       
 with a:  # magic trailing comma
```

I also tried inverting the order (`optional_parentheses(fits_expanded(...))`) like we talked about, but it then doesn't wrap the f-string case:

```diff
-msg = lambda x: (                                                                                       
-    f"this is a very very very very long lambda value {x} that doesn't fit on a single line"            
-)                                                                                                       
+msg = lambda x: f"this is a very very very very long lambda value {x} that doesn't fit on a single line"
```

That seems to be the only test failure, though.

Reply:
ah sorry, I meant `parenthesize_if_expands`

We can't use `fits_expanded` because it's causing this weird wrapping here where the binary expression continues on the same line

```patch
         # If it has been indicated that the user wants the interpolator, construct
         # an anonymous function to handle units and sanitize IO
         if return_interpolator:
-            return (
-                lambda x: np.exp(cs(np.log(x.to(u.MeV).value))) * u.MeV * u.cm**2 / u.g
-            )
+            return lambda x: np.exp(
+                cs(np.log(x.to(u.MeV).value))
+            ) * u.MeV * u.cm**2 / u.g
 
         return (
             energies,
```